services/api/mw2pdf/utils/validation.js

```diff
@@ -0,0 +1,31 @@
+import Joi from 'joi'
+
+export function assertValidPassthroughParameters(passthroughParameters) {
+  Joi.assert(
+    passthroughParameters,
+    Joi.string()
+      .custom(
+        (value, helpers) => {
+          JSON.parse(value)
+          return value
+        },
+        'is a valid JSON string',
+      ),
+  )
+  Joi.assert(
+    JSON.parse(passthroughParameters),
+    Joi.object()
+      .pattern(
+        /\w+/,
+        Joi.any().custom(
+          (value, helpers) => {
+            if (typeof value === 'object' && value !== null) {
```

Comment:
We expect every key's value to be an object and not null?

Reply:
I'll make this a named helper method -- yes.